h2o-algos/src/main/java/hex/ensemble/StackedEnsembleModel.java

```diff
@@ -387,26 +533,66 @@ void checkAndInheritModelProperties() {
         }
 
-        // In GLM, we get _family instead of _distribution.
-        // Further, we have Family.binomial instead of DistributionFamily.bernoulli.
-        // We also handle DistributionFamily.AUTO in distributionFamily()
-        //
-        // Hack alert: DRF only does Bernoulli and Gaussian, so only compare _domains.length above.
-        if (! (aModel instanceof DRFModel) && distributionFamily(aModel) != distributionFamily(this))
-          Log.warn("Base models are inconsistent; they use different distributions: "
-                  +distributionFamily(this)+" and: "+distributionFamily(aModel) +". Is this intentional?");
-
-        // TODO: If we're set to DistributionFamily.AUTO then GLM might auto-conform the response column
-        // giving us inconsistencies.
+        if (inferredDistributionFromFirstModel) {
+          // In GLM, we get _family instead of _distribution.
+          // Further, we have Family.binomial instead of DistributionFamily.bernoulli.
+          // We also handle DistributionFamily.AUTO in distributionFamily()
+          //
+          // Hack alert: DRF only does Bernoulli and Gaussian, so only compare _domains.length above.
+          // TODO: If we're set to DistributionFamily.AUTO then GLM might auto-conform the response column
+          // giving us inconsistencies.
+
+          // Check inferred params and if they differ fallback to basic distribution of model category
+          if (!(aModel instanceof DRFModel) && distributionFamily(aModel) == distributionFamily(this)) {
+            boolean sameParams = true;
+            switch (_parms._metalearner_parameters._distribution) {
+              case custom:
+                sameParams = _parms._metalearner_parameters._custom_distribution_func
+                        .equals(aModel._parms._custom_distribution_func);
+                break;
+              case huber:
+                sameParams = _parms._metalearner_parameters._huber_alpha == aModel._parms._huber_alpha;
+                break;
+              case tweedie:
+                sameParams = _parms._metalearner_parameters._tweedie_power == aModel._parms._tweedie_power;
+                break;
+              case quantile:
+                sameParams = _parms._metalearner_parameters._quantile_alpha == aModel._parms._quantile_alpha;
+                break;
+            }
+            if ((aModel instanceof GLMModel) &&
+                    (Metalearners.getActualMetalearnerAlgo(_parms._metalearner_algorithm) == Metalearner.Algorithm.glm) &&
+                    !((GLMModel.GLMParameters) _parms._metalearner_parameters)._link.equals(((GLMModel) aModel)._parms._link)) {
+              if (inferredFromGLM) {
+                sameParams = false;
+              } else {
+                inheritFamilyAndParms(aModel._parms);
+                inferredFromGLM = true;
+              }
+            }
```

Comment:
maybe it would have been more clear to track `firstGLM` instead of `inferredFromGLM`.
This way you can get use similar logic for `sameParams` as for others, sth approx. like:
```
if aModel instanceof GLMModel and meta_algo == glm:
   if firstGLM == null:
      firstGLM = aModel
      inherit...
   else:
      sameParams = _parms._metalearner_parameters._link == aModel._parms._link
```

it has the advantage of being more in line with the code above, and a bit easier to understand imo. wdyt?

Reply:
I agree